src/Service/Authorization/AuthorizationResolver.cs

```diff
@@ -139,38 +139,49 @@ public bool AreColumnsAllowedForOperation(string entityName, string roleName, Co
             // Columns.Count() will never be zero because this method is called after a check ensures Count() > 0
             Assert.IsFalse(columns.Count() == 0, message: "columns.Count() should be greater than 0.");
 
-            OperationMetadata operationToColumnMap = EntityPermissionsMap[entityName].RoleToOperationMap[roleName].OperationToColumnMap[operation];
+            //OperationMetadata operationToColumnMap = EntityPermissionsMap[entityName].RoleToOperationMap[roleName].OperationToColumnMap[operation];
+            if (!EntityPermissionsMap[entityName].RoleToOperationMap.TryGetValue(roleName, out RoleMetadata? roleMetadata) && roleMetadata is null)
```

Comment:
What would be the scenarios in which this lookup could fail? Why didn't we detect this earlier?

Reply:
I believe i did this out of abundance of caution. I couldnt identify a case this fails after i finalized for publishing the pr.

Reply:
Great, thanks for making our code failsafe.